src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -174,7 +174,7 @@ def __init__(
         repository_pex: Pex | None = None,
         additional_args: Iterable[str] = (),
         pex_path: Iterable[Pex] = (),
-        apply_requirement_constraints: bool = True,
+        apply_requirement_constraints: bool = False,
```

Comment:
There is at least one other `PexRequest` call that should set this: https://github.com/pantsbuild/pants/blob/adbc61da8a468705feaa54f456f5193a91871ef7/src/python/pants/backend/python/typecheck/mypy/rules.py#L163-L178 (...should probably also consume the `repository.pex`...?).

Reply:
We will already have created `requirements_pex` using `PexFromTargets` - this here only converts that Pex zipapp into a Venv. No resolution should happen, iiuc. @jsirois does that sound right?